### PR TITLE
Refactor InstrumentedCall to Call in instrument package

### PR DESCRIPTION
**Pull request by robskillington**
To reduce public API surface area from the base `tally` package.

### CODE REVIEW

File: instrument/call.go [hunk 1]
```diff
@@ -18,34 +18,34 @@
...
+	timingSuffix      = "latency"
 )
 
-// NewInstrumentedCall returns an InstrumentedCall with the given name
-func NewInstrumentedCall(scope Scope, name string) InstrumentedCall {
-	return &instrumentedCall{
-		error:   scope.Tagged(map[string]string{_resultType: _resultTypeError}).Counter(name),
-		success: scope.Tagged(map[string]string{_resultType: _resultTypeSuccess}).Counter(name),
-		timing:  scope.SubScope(name).Timer(_timingFormat),
+// NewCall returns an instrumented call with the given name
```
Comment (prateek) [inline instrument/call.go]: nit: NewCall returns a Call ...

File: instrument/call.go [hunk 2]
```diff
@@ -18,34 +18,34 @@
...
+	scope   tally.Scope
+	success tally.Counter
+	error   tally.Counter
+	timing  tally.Timer
 }
 
-// Exec executes the given block of code, and records whether it succeeded or
-// failed, and the amount of time that it took
-func (c *instrumentedCall) Exec(f ExecFn) error {
+func (c *call) Exec(f ExecFn) error {
```
Comment (jeromefroe) [inline instrument/call.go]: nit: perhaps add back the comment for the function?

File: instrument/types.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package instrument
+
+// ExecFn is a Call executable
+type ExecFn func() error
+
+// Call allows tracking the success, errors, and timing of functions
```
Comment (jeromefroe) [inline instrument/types.go]: nit: s/success/successes

File: instrument/call.go (hunk 1 above)
Comment (robskillington) [reply instrument/call.go]: Sure thing.

File: instrument/call_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+package instrument
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/uber-go/tally"
```
Comment (prateek) [inline instrument/call_test.go]: nit: import ordering ~ move this to a separate block above 3rd party imports

File: instrument/types.go [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package instrument
+
+// ExecFn is a Call executable
```
Comment (prateek) [inline instrument/types.go]: maybe reword to 
``` // ExecFn is an executable function. ```
